content/posts/building-images-with-img.md

```diff
@@ -0,0 +1,255 @@
+---
+title: Building Container Images with img
+author:
+  name: "Matt Elgin"
+date: 2019-10-25T00:00:00-00:00
+showDate: true
+tags: ["kubernetes","img","docker","jenkins","containers","google kubernetes engine","workload identity"]
+draft: true
+---
+
+## Why use `img` for container builds?
+As more organizations turn to containers and Kubernetes to manage their CI/CD workloads, numerous strategies have emerged to handle the actual building of container images within these containerized environments. However, each of these approaches have not been without their security drawbacks (see [Kurt Madel](/authors/kurt-madel/)'s recent post on ["Securely Building Container Images on Kubernetes"](/posts/build-continaer-images/) for a rundown of these approaches and their security implications.)
+
+The current choice for many teams is Google's [kaniko](https://github.com/GoogleContainerTools/kaniko) project. This is a good option for many organizations (especially when combined with Kurt's recommendations for `PodSecurityPolicies`). However, it's worth noting that while `kaniko` itself does not require running as root, it will [require that privilege for most significant container building](https://github.com/GoogleContainerTools/kaniko#security). While this is an acceptable caveat for some organizations, it can be a deal-breaker for others.
+
+Fortunately, efforts have been underway to create an unprivileged, non-root container builder. Jessie Frazelle introduced [`img`](https://github.com/genuinetools/img), one such tool, in her blog post ["Building Container Images Securely on Kubernetes"](https://blog.jessfraz.com/post/building-container-images-securely-on-kubernetes/). I'll leave the details of design approach and usage to her post and the GitHub project page, but there are a few points I'll highlight here:
+
+1. `img` is intended to run as an arbitrary, non-root user 1000.
```

Comment:
You mention 'arbitrary' but then specifically mention user 1000 - but I guess it doesn't have be 1000 - so maybe "non-root user such as 1000."